os/board/common/crashdump.c

```diff
@@ -156,6 +234,57 @@ static int ramdump_via_uart(void)
 		}
 	}
 
+	/* Will dump Filesystem contents */
+	up_lowputc('F');
+
+	/* Receive handshake string from HOST */
+	do {
+		host_buf[0] = up_getc();
+	} while (host_buf[0] != target_str[0]);
+
+	for (i = 1; i < strlen(target_str);) {
+		if ((ch = up_getc()) != -1) {
+			host_buf[i] = ch;
+			i++;
+		}
+	}
+
+	if (strncmp(host_buf, FSDUMP_HANDSHAKE_STRING, strlen(FSDUMP_HANDSHAKE_STRING) + 1) != 0) {
+		/* Send NAK */
+		up_lowputc('N');
+	}
+
+filesystem_dump:
+	/* Send ACK */
+	up_lowputc('A');
+
+	x = get_userfs_address_size(&userfs_start, &size);
+	if (x < 0) {
+		return -1;
```

Comment:
Since this line & line no. 146 return the same error, can we differentiate amongst them to indicate why exactly the dump fails in dump_via_uart function ??

Reply:
Since we are in a crashed state:- 
1. Will the debug logs be any useful?
2. HOST has acquired tty connection to the TARGET, any debug logs will not be printed out to the console.

However, before Line 146, since the tool has not yet started, we could add a debug line to inform that the low level APIs are not available for communication.

Reply:
Yes I think that would be helpful. I feel debugging at programmer level regarding why ramdump is failing will also be easier.

Reply:
Added